app/client/cypress/e2e/Regression/ClientSide/Anvil/AnvilCopyPaste_spec.ts

```diff
@@ -0,0 +1,154 @@
+      // so the URL should be different from the URL before copy
+      cy.get("@widgetSelectURLBeforeCopy").should("not.eq", cy.url());
+
+      // paste into section
+      agHelper.GetNClick(
+        anvilLocators.anvilWidgetNameSelectorFromEntityExplorer("Section1"),
+      );
+      cy.get("body").type(`{${modifierKey}}v`);
+      // since we pasted in section, a new zone and a new copied button should be created
+      agHelper.AssertElementLength(
+        anvilLocators.anvilWidgetTypeSelector(anvilLocators.ZONE),
+        3,
+      );
+      agHelper.AssertElementLength(
+        anvilLocators.anvilWidgetTypeSelector(anvilLocators.WDSBUTTON),
+        3,
+      );
+      // and no new section should be created and space redistribution should be [6, 6]
+      agHelper.AssertElementLength(
+        anvilLocators.anvilWidgetTypeSelector(anvilLocators.SECTION),
+        2,
+      );
+      anvilLayout.sections.verifySectionDistribution("Section1", [6, 6]);
+
+      // paste into zone
+      agHelper.GetNClick(anvilLocators.anvilWidgetNameSelector("Zone1"));
+      cy.get("body").type(`{${modifierKey}}v`);
+      // since we pasted in zone, a new copied button should be created, no new zone or section should be created
+      agHelper.AssertElementLength(
+        anvilLocators.anvilWidgetTypeSelector(anvilLocators.WDSBUTTON),
+        4,
+      );
+      agHelper.AssertElementLength(
+        anvilLocators.anvilWidgetTypeSelector(anvilLocators.ZONE),
+        3,
+      );
+      agHelper.AssertElementLength(
+        anvilLocators.anvilWidgetTypeSelector(anvilLocators.SECTION),
+        2,
+      );
+    });
+
+    it("2. Checks if the modal widget is pasted into the main canvas", () => {
+      // drop a modal widget
+      anvilLayout.dnd.DragDropNewAnvilWidgetNVerify(
+        anvilLocators.WDSMODAL,
+        10,
+        10,
+        {
+          skipWidgetSearch: true,
+          dropTargetDetails: {
+            dropModal: true,
+          },
+        },
+      );
+
+      // copy the modal widget
+      cy.get("body").type(`{${modifierKey}}c`);
+
+      // press escape to close the modal
+      agHelper.PressEscape();
+
+      // select the canvas and paste into canvas
+      agHelper.GetNClick(anvilLocators.mainCanvasSelector);
+      cy.get("body").type(`{${modifierKey}}v`);
+      // since we pasted in canvas, a new copied modal should be created
+      agHelper.AssertElementLength(
+        anvilLocators.anvilWidgetNameSelectorFromEntityExplorer("Modal1Copy"),
+        1,
+      );
+
+      // close the modal
+      agHelper.PressEscape();
+    });
+
+    it("3. Checks if no widget is selected, the main canvas is the destined parent", () => {
+      // copy the button widget
+      agHelper.GetNClick(anvilLocators.anvilWidgetNameSelector("Button1"));
+      cy.get("body").type(`{${modifierKey}}c`);
+
+      // deselect widgets by clicking on the canvas
+      cy.get(`${anvilLocators.mainCanvasSelector} > div`).click();
```

Comment:
Can we use `anvilLocators.mainCanvasSelector` as a selector? Why do we need the div here? 

```suggestion
      cy.get(`${anvilLocators.mainCanvasSelector}`).click();
```

Reply:
Updated. I saw it in one of ashok's tests. So copied from there.